packages/iris-grid/src/IrisGridTableModelTemplate.ts

```diff
@@ -1625,7 +1622,11 @@ class IrisGridTableModelTemplate<
   }
 
   isKeyColumn(x: ModelIndex): boolean {
-    return x < (this.inputTable?.keyColumns.length ?? 0);
+    if (x < 0 || x >= this.columnCount) {
+      return false;
+    }
+
```

Comment:
I don't think this check is necessary - we should never be passed an `x` that is outside of the column count range.